wagtail/admin/tests/viewsets/test_image_presence.py

```diff
@@ -0,0 +1,49 @@
+import io
+from urllib.parse import urljoin
+
+import requests
+from bs4 import BeautifulSoup
+from django.test import TestCase
+
+from wagtail.test.utils import WagtailTestUtils
+
+# This test was made in order to test the Dynamic Image Serve View.
+# For this test you will need to change your "mysite" in order for its home_page.html to show
+# an image.
+# To do that, you need to ensure that you have (it can be dynamic) serve view in the template and update
+# the urls.py. Problably an addition of "image = models.ForeignKey( get_image_model(), ...)"
+# field to the HomePage class will also be needed
+
+
+class TestImagePresence(WagtailTestUtils, TestCase):
+    def test_image_presence(self):

```

Comment:
Issue: This test case feels rather complex considering what it's testing. You should be able to create an image URL for an existing image, and then check the response looks like an image.

There are already tests for the image serve view, so this would make sense there. I'm also not sure what this is testing that the other tests aren't already.

Reply:
I added test_image_presence.py because i saw that all of the tests that were in the main code were passing while the bug was still present. Pardon me if i am mistaken but it seems none of them were testing the Dynamic Serve View. To resolve that, I made this test and i made sure that it passed  with the fix and it failed without it. All of this code stands for the fact that i had some troubles replicating the bug on a test, hence all of the code. With that being said, you prefer i reduce the complexety of the test? I could try to do that but i think it's presence is necessary to verify the bug fix. I am new to wagtail and open source code in general, so I apologize in advance for my lack of experience, and especially my late reply.

Thank you for your time,
Afonso Matos

Reply:
There are definitely tests for the serve view here: https://github.com/wagtail/wagtail/blob/252bae9129f9167d371d65392ae95b2db1961de0/wagtail/images/tests/tests.py#L336

You should be able to adapt what you have to work with the setup there, which shows you how to find the serve URL for a given image. Your tests seem to also assume you have a separate Wagtail instance running on port `8000`, which is definitely not right - you'll need to be using the test client.

To write the test, I'd recommend something test-driven. Undo the fix in `serve.py`, and then write a test case which fails, and thus exercises the bug. Fix the bug, and the test case should pass. _That_ is the perfect test :slightly_smiling_face: